lib/components/errors/connectivity_error_component.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:Medito/components/components.dart';
+import 'package:Medito/constants/constants.dart';
+import 'package:Medito/providers/providers.dart';
+import 'package:Medito/utils/utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+
+class ConnectivityErrorComponent extends ConsumerStatefulWidget {
+  const ConnectivityErrorComponent({super.key});
+
+  @override
+  ConsumerState<ConnectivityErrorComponent> createState() =>
+      _ConnectivityErrorComponentState();
+}
+
+class _ConnectivityErrorComponentState
+    extends ConsumerState<ConnectivityErrorComponent> {
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      ref.read(audioPlayPauseStateProvider.notifier).state =
+          PLAY_PAUSE_AUDIO.PAUSE;
+    });
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _refreshAPI();
+    super.dispose();
+  }
+
+  void _refreshAPI() {
+    var location = GoRouter.of(context).location;
+    var id = location.split('/');
```

Comment:
The .last() should be here I think right otherwise me get an array?

Reply:
` ref.read(meditationsProvider(meditationId: int.parse(id.last)));`

I was doing something like this. But I think instead of doing this I can get the last element here
`var id = location.split('/').last;`